packages/@aws-cdk/aws-apigatewayv2/lib/common/api-mapping.ts

```diff
@@ -11,3 +17,107 @@ export interface IApiMapping extends IResource {
    */
   readonly apiMappingId: string;
 }
+
+/**
+ * Properties used to create the ApiMapping resource
+ */
+export interface ApiMappingProps {
+  /**
+   * Api mapping key. The path where this stage should be mapped to on the domain
+   * @default - undefined for the root path mapping.
+   */
+  readonly apiMappingKey?: string;
+
+  /**
+   * The Api to which this mapping is applied
+   */
+  readonly api: IApi;
+
+  /**
+   * custom domain name of the mapping target
+   */
+  readonly domainName: IDomainName;
+
+  /**
+   * stage for the ApiMapping resource
+   *
+   * @default - Default stage of the passed API
+   */
+  readonly stage?: IStage;
+}
+
+/**
+ * The attributes used to import existing ApiMapping
+ */
+export interface ApiMappingAttributes {
+  /**
+   * The API mapping ID
+   */
+  readonly apiMappingId: string;
+}
+
+/**
+ * Create a new API mapping for API Gateway API endpoint.
+ * @resource AWS::ApiGatewayV2::ApiMapping
+ */
+export class ApiMapping extends Resource implements IApiMapping {
+  /**
+   * import from API ID
+   */
+  public static fromApiMappingAttributes(scope: Construct, id: string, attrs: ApiMappingAttributes): IApiMapping {
+    class Import extends Resource implements IApiMapping {
+      public readonly apiMappingId = attrs.apiMappingId;
+    }
+    return new Import(scope, id);
+  }
+  /**
+   * ID of the API Mapping
+   */
+  public readonly apiMappingId: string;
+
+  /**
+   * API Mapping key
+   */
+  public readonly mappingKey?: string;
+
+  constructor(scope: Construct, id: string, props: ApiMappingProps) {
+    super(scope, id);
+
+    let stage = props.stage;
+    if (!stage) {
+      if (props.api instanceof HttpApi) {
+        if (props.api.defaultStage) {
+          stage = props.api.defaultStage;
+        } else {
+          throw new Error('stage is required if default stage is not available');
+        }
+      } else {
+        throw new Error('stage is required for WebSocket API');
+      }
+    }
```

Comment:
So the `api` prop would still be required since currently `api` is not available on `IStage` since we moved it to subclass `IHttpStage` and `IWebSocketStage`. Now given that, we can keep this implementation because it provided convenience for HttpApi users (though agree this is not the cleanest of solutions but maybe in future it can be clear when we have defaultStage for WS apis as well). Thoughts?

Reply:
So now we can go from `IHttpStage` -> `IHttpApi` but can't go from `IStage` -> `IApi`?  Whoops.
I think we should somehow get both but let's not try to solve this now in this PR.

Leave it as such.